yarn-project/end-to-end/src/e2e_deploy_contract.test.ts

```diff
@@ -128,4 +128,27 @@ describe('e2e_deploy_contract', () => {
       portalContract.toString(),
     );
   });
+
+  it('it should not deploy a contract which failed the public part of the execution', async () => {
+    // This test requires at least another good transaction to go through in the same block as the bad one.
+    // I deployed the same contract again but it could really be any valid transaction here.
```

Comment:
Let's add an assertion that both txs landed on the same block. We may need to simulate them before sending them simultaneously, as we're doing in the block_building e2e.

Reply:
Good point. But checking that it doesn't end up in a block won't guarantee that you have reproduced the conditions for the bug in this test.

I can think of a few alternatives:

- Call [`updateConfig`](https://github.com/AztecProtocol/aztec-packages/blob/b30839e9e5b88124443d35140f84610bbc0a7855/yarn-project/sequencer-client/src/sequencer/sequencer.ts#L59) on the sequencer (after the setup is done) to set `minTxsPerBlock` to 2.
- Add a jest spy on the p2p client to check that the two txs were returned simultaneously.
- Hook up to the logger via [`onLog`](https://github.com/AztecProtocol/aztec-packages/blob/b30839e9e5b88124443d35140f84610bbc0a7855/yarn-project/foundation/src/log/logger.ts#L64) and check the sequencer logs to ensure it picked up both txs at the same time.

Reply:
I've gone with the first approach, updating the sequencer config.